depolarizer.py

```diff
@@ -4,32 +4,66 @@
 def width(file_i):
     return cv.imread(file_i).shape[1]
 
-def depolarizer(file_i, file_o=None, crop=None, axis=0, res_o=1000):
-    img_i = cv.imread(file_i, cv.IMREAD_COLOR)
-
-    if crop is not None:
-        x = round(crop["x"]) + np.arange(round(crop["width"]))
-        y = round(crop["y"]) + np.arange(round(crop["height"]))
-        img_i = img_i[y[:, None], x]
-    assert(img_i.shape[0] == img_i.shape[1])
-
-    dim_i = img_i.shape[0]
-    res_i = np.sqrt(2) * dim_i/2
-
-    seq = np.arange(res_o, dtype=np.float32)
-    pix_o = np.meshgrid(seq, np.flip(seq))
-    r = res_i * np.exp(2*np.pi * (pix_o[1] / res_o - 1))
-    angle = 2*np.pi * pix_o[0] / res_o - axis * np.pi / 180
-    map_x = r * np.cos(angle) + dim_i/2
-    map_y = r * np.sin(angle) + dim_i/2
-    img_o = cv.remap(img_i, map_x, map_y, cv.INTER_LINEAR)
-
-    if file_o is None:
-        return img_o
-    cv.imwrite(file_o, img_o)
+class depolarizer:
+    def __init__(self, file_i, crop=None):
+        # receives the input file, crops it and saves it as an attribute. 
+        self._img_i = cv.imread(file_i, cv.IMREAD_COLOR)
+        if crop is not None:
+            x = round(crop["x"]) + np.arange(round(crop["width"]))
+            y = round(crop["y"]) + np.arange(round(crop["height"]))
+            self._img_i = self._img_i[y[:, None], x]
+
+        assert(self._img_i.shape[0] == self._img_i.shape[1])
+
+    def __map(self, map_x, map_y, file_o = None):
+        # receives self._img_i and remaps it into img_o with two given functions (map_x, map_y). 
+        img_o = cv.remap(self._img_i, map_x, map_y, cv.INTER_LINEAR)
+
+        if file_o is None:
+            return img_o
+        cv.imwrite(file_o, img_o)
+
+    def to_cartesian(self, file_o=None, axis=0, res_o=1000):
+      
+        # generate the array of coordinates for the output image. 
+        seq = np.arange(res_o, dtype=np.float32)
+        pix_o = np.meshgrid(seq, seq)
+        
+        # auxiliar variables
+        pix_o[0] -= res_o/2 
+        pix_o[1] -= res_o/2
+        res_i = self._img_i.shape[0]
+        diag_o = np.sqrt(2) * res_o/2
+        r = np.log( np.sqrt(pix_o[1]**2 + pix_o[0]**2)*np.exp(2*np.pi)/diag_o)
```

Comment:
Wouldn't it be better to take `2*np.pi` out of the logarithm?

Reply:
Noted!